src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -198,6 +198,10 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
             {
                 ThrowHelper.ThrowInvalidProgramException();
             }
+            if (result == CorJitResult.CORJIT_IMPLLIMITATION)
+            {
+                throw new RequiresRuntimeJitException("JIT implementation limitation");
```

Comment:
To avoid build break in runtimelab:

```suggestion
#if READYTORUN
                throw new RequiresRuntimeJitException("JIT implementation limitation");
#else
                ThrowHelper.ThrowInvalidProgramException();
#endif
```